### PR TITLE
[Improve][Connector-V2][Kafka]Unified exception for Kafka source and sink connector

### PR DESCRIPTION
**Pull request by TaoZex**
## Purpose of this pull request
https://github.com/apache/incubator-seatunnel/issues/3515
Unified exception for Kafka source and sink connector

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)

### CODE REVIEW

**Review by Hisoka-X — DISMISSED** [review]
LGTM

**Comment by TaoZex** [comment]
> LGTM

Thanks for your review.Have a great weekend!

File: seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/sink/KafkaSinkWriter.java [hunk 1]
```diff
@@ -124,7 +127,8 @@ public void close() {
         try (KafkaProduceSender<?, ?> kafkaProduceSender = kafkaProducerSender) {
             // no-opt
         } catch (Exception e) {
-            throw new RuntimeException("Close kafka sink writer error", e);
+            throw new KafkaConnectorException(KafkaConnectorErrorCode.CLOSE_SINK_WRITER_FAILED,
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/sink/KafkaSinkWriter.java]: Common.WRITER_OPERATION_FAILED is better, remove useless error code.

**Review by TyrantLucifer — APPROVED** [review]
+1

**Review by wuchunfu — APPROVED** [review]

**Pull request merged**